Branch-SDK/Branch-SDK/Branch.m

```diff
@@ -1004,35 +714,16 @@ - (void)applyReferralCode:(NSString *)code andCallback:(callbackWithParams)callb
     }
     
     
-    BNCServerRequest *req = [[BNCServerRequest alloc] init];
-    req.tag = REQ_TAG_APPLY_REFERRAL_CODE;
-    NSMutableDictionary *post = [NSMutableDictionary dictionaryWithObjects:@[code,
-                                                                             [BNCPreferenceHelper getIdentityID],
-                                                                             [BNCPreferenceHelper getSessionID],
-                                                                             [BNCPreferenceHelper getDeviceFingerprintID]]
-                                                                   forKeys:@[BRANCH_DATA_KEY_REFERRAL_CODE,
-                                                                             BRANCH_DATA_KEY_IDENTITY_ID,
-                                                                             BRANCH_DATA_KEY_SESSION_ID,
-                                                                             BRANCH_DATA_KEY_DEVICE_FINGERPRINT_ID]];
-    req.postData = post;
-    req.callback = ^(BNCServerResponse *response, NSError *error) {
-        if (error) {
-            if (callback) {
-                callback(nil, error);
-            }
-            return;
-        }
+    BranchApplyReferralCodeRequest *req = [[BranchApplyReferralCodeRequest alloc] initWithCode:code callback:callback];
+    [self.requestQueue enqueue:req];
+    [self processNextQueueItem];
+}
 
-        if (![response.data objectForKey:BRANCH_DATA_KEY_REFERRAL_CODE]) {
-            error = [NSError errorWithDomain:BNCErrorDomain code:BNCInvalidReferralCodeError userInfo:@{ NSLocalizedDescriptionKey: @"Referral code is invalid - it may have already been used or the code might not exist" }];
-        }
-        
-        if (callback) {
-            callback(response.data, error);
-        }
-    };
 
-    [self.requestQueue enqueue:req];
+#pragma mark - Logging
+- (void)log:(NSString *)log {
+    BranchLogRequest *request = [[BranchLogRequest alloc] initWithLog:log];
+    [self.requestQueue enqueue:request];
```

Comment:
Perhaps I was the one misunderstanding. I didn't know why we had two queues, so I just merged them. I didn't like that the PreferenceHelper had it's own queue, it felt strange for a preference helper. I could make a separate one here, though.